### ISSUE TITLE
[Feature Request] Randomize Link for public Collection

### ISSUE THREAD

**Issue by cardi-m** [linkwarden/linkwarden#248]
The public collection Link starts actually with 1 and counts up. When i know one link it is easy to try others. It would be nice to have a randomized link. (By default or an option to choose by creating the link)

**Comment by ser** [comment]
IMHO it's extremely important as current setup is clearly very insecure. And easy to deploy.

**Comment by daniel31x13** [comment]
> When i know one link it is easy to try others.

Correct, but in order to “try others” you’ll need to be authenticated.

I don’t see any problem with that, could you please elaborate what could possibly go wrong?

**Comment by cardi-m** [comment]
> > When i know one link it is easy to try others.
> 
> Correct, but in order to “try others” you’ll need to be authenticated.
> 
When i have more than one public collection and they just count up i just have to try another digit an the chance is high to find them, maybe i have a collection for a group public (because bot everybody wants to click a bew account) is the chance lower to find it when i have a bigger ammount of diggits.



> 
> I don’t see any problem with that, could you please elaborate what could possibly go wrong?

**Comment by daniel31x13** [comment]
But it's ok if your public names are visible/findable by the public (hence the name), isn't it?

I'll leave this issue open anyway to see if anyone else had this concern but for now, I don't think that's too big of an issue...

**Comment by ser** [comment]
`public` is not equal to `findable` - it's a big issue.

**Comment by lo48576** [comment]
It sounds to me like you are trying to do some kind of "security by obscurity"... which is known to be a bad practice.
URI of a public collection could be non-findable directly from URL, but there is no other protection from the collection to be found, for example someone accidentally shared it publicly, browser or addons are reporting the URL to the search engine, etc.
If your concern is "security", all you need is authentication or authorization, not randomized URIs.

Anyway, "unlisted" share is implemented by some services (e.g. "Unlisted" visibility on YouTube) so it would actually be useful in some situation while it has almost nothing to do with the security.

**Comment by ser** [comment]
so let's make a short summary of your comment, @lo48576 - you have just wrote that google is doing "security by obscurity".

**Comment by lo48576** [comment]
It's users' fault to believe unlisted URLs to be "secure" without additional auth.

Unlisted resources are visible to *possibly* limited users, but it just reduces the exposure and fundamentally does not limit user accesses without additional authorization.
If you need to limit accesses to the listed *specific* limited users, unlisted URLs are not the right means.
Calling the former (unable to make things unlisted) "very insecure" sounds somewhat inconsistent or too strong.

If your intention is just reducing exposure of the URLs, then it basically does not improve the security. Even worse, people will feel secure while it is actually not secure.
On the other hand, if your concern is security, you'll need the latter (auth), not the former (unlisted URLs).

**Comment by ser** [comment]
so let's make a short summary of your comment, @lo48576 , Google forces users to get faulted believing their unlisted URI's are secure without additional auth.

**Comment by daniel31x13** [comment]
Hey I'm closing this due to the small demand. Sorry.
